samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/SemanticChatMemoryExtractor.cs

```diff
@@ -67,39 +73,30 @@ internal static class SemanticChatMemoryExtractor
     /// Extracts the semantic chat memory from the chat session.
     /// </summary>
     /// <param name="memoryName">Name of the memory category</param>
-    /// <param name="kernel">The semantic kernel.</param>
     /// <param name="context">The SKContext</param>
     /// <param name="options">The prompts options.</param>
     /// <returns>A SemanticChatMemory object.</returns>
     internal static async Task<SemanticChatMemory> ExtractCognitiveMemoryAsync(
         string memoryName,
-        IKernel kernel,
-        SKContext context,
-        PromptsOptions options)
+        SKContext memoryExtractionContext,
+        PromptsOptions options,
+        ISKFunction extractMemorySKFunc,
+        int memoryPromptTokenCount)
     {
-        if (!options.MemoryMap.TryGetValue(memoryName, out var memoryPrompt))
+        if (!options.MemoryTypes.Contains(memoryName))
         {
             throw new ArgumentException($"Memory name {memoryName} is not supported.");
         }
 
         // Token limit for chat history
-        var tokenLimit = options.CompletionTokenLimit;
         var remainingToken =
-            tokenLimit -
+            options.CompletionTokenLimit -
             options.ResponseTokenLimit -
-            Utilities.TokenCount(memoryPrompt); ;
+            memoryPromptTokenCount;
 
-        var memoryExtractionContext = Utilities.CopyContextWithVariablesClone(context);
         memoryExtractionContext.Variables.Set("tokenLimit", remainingToken.ToString(new NumberFormatInfo()));
-        memoryExtractionContext.Variables.Set("memoryName", memoryName);
-        memoryExtractionContext.Variables.Set("format", options.MemoryFormat);
-        memoryExtractionContext.Variables.Set("knowledgeCutoff", options.KnowledgeCutoffDate);
 
-        var completionFunction = kernel.CreateSemanticFunction(memoryPrompt);
-        var result = await completionFunction.InvokeAsync(
-            context: memoryExtractionContext,
-            settings: CreateMemoryExtractionSettings(options)
-        );
+        var result = await extractMemorySKFunc.InvokeAsync(memoryExtractionContext);
```

Comment:
Passing the tokenLimit in the context isn't enough to manage the tokens, you have to specify new CompleteRequestSettings as a second parameter, similar to:

```
chatContext = await completionFunction.InvokeAsync(
            context: chatContext,
            settings: this.CreateChatResponseCompletionSettings()
        );
```

The `CreateChatResponseCompletionSettings` function is defined in ChatSkill

Reply:
Good catch, forgot to update those;